lib/strings.nix

```diff
@@ -3,6 +3,7 @@
 let
 
 inherit (builtins) length;
+inherit (lib.lists) intersperse;
```

Comment:
Are there parts of the nixpkgs code that depend on this directly? If so, it would be best to fix those up. As for external uses, I think it would be good to add a remark that this is deprecated or possibly even add a wrapper function that logs out a warning so that we can actually ditch this in some while.

Reply:
It seems that only `lib.strings.concatStringsSep` depends on `intersperse` in the case that `builtins.concatStringSep` doesn't exist.

Are you suggesting that we keep `lib.strings.intersperse` and put it behind `lib.trivial.warn`?

Reply:
As per whether or not any other code in nixpkgs depends on `lib.strings.intersperse`, not that I can tell. Anywhere else using `intersperse` seems to specify `lib.intersperse` and not `lib.strings.intersperse` directly.

Reply:
I went through and have confirmed nothing depends on `lib.strings.intersperse`.